### ISSUE TITLE
How to measure compressed video size

### ISSUE THREAD

**Issue by Zayn-Rekhi** [InterDigitalInc/CompressAI#191]
## How to measure compressed video size?
Hi guys, I really appreciate all of your video compression work. I cannot thank you guys enough. I was wondering how I can measure the size of my compressed video. Any help or suggestions would be greatly appreciated :) Thanks.

**Comment by YodaEmbedding** [comment]
Two ways:

- Generate a bitstream or file and check its size.
- During training, compute the _differentiable_ rate via the method below.

Just like with the VAE image compression models, the rate in bpp can be measured by via the cross-entropy between the actual symbols and their "predicted" distributions. Whenever the predicted distribution matches, the bit cost for encoding that symbol is low. The sum of the log summing the log-likelihood.

Since these models minimize `loss = R + lambda D`, you may find the calculation for rate `R` (measured in bpp) in the `RateDistortionLoss` class in `examples/train_video.py`.

Note: This question is best asked in [Discussions](https://github.com/InterDigitalInc/CompressAI/discussions).

**Comment by Zayn-Rekhi** [comment]
I apologize for writing my question, I will make sure to write it in discussion moving forward. I liked your first suggestion and looking at the evaluation code I think that this has already been implemented. However, when I check the size of the binary file it is much, much smaller than the mp4 file produced by h265. Hence, I think I did something wrong. What could I have done wrong?

**Comment by YodaEmbedding** [comment]
Perhaps one of these:
 https://github.com/InterDigitalInc/CompressAI/issues/190#issuecomment-1368918788
